<!-- 
	Node launch file for bta_tof_driver.
	Display depth from compresed U16C1 row data
	Thanks to Jon Azpiazu (Jon.Azpiazu@sintef.no)

	See http://www.ros.org/wiki/bta_tof_driver for more information.
-->
<launch>
  <!-- Nodelet manager for this pipeline -->
  <node pkg="nodelet" type="nodelet" args="manager"
        name="record_player_manager" output="screen"/>
  
  <!-- The depth image is already rectified and registered
       to the camera optical frame, but stored in mm; convert it
       to meters -->
  <node pkg="nodelet" type="nodelet" name="metric_rect" 
        args="load depth_image_proc/convert_metric record_player_manager --no-bond">

	<remap from="image_raw" to="/bta_tof_driver_1/tof_camera/decompressedDepth"/>
	<remap from="image" to="/bta_tof_driver_1/tof_camera/depth/image"/>
  </node>
  
  <node pkg="image_transport" type="republish" name="image_convert" 
  	args="compressedDepth">
	<remap from="in" to="/bta_tof_driver_1/tof_camera/compressedDepth"/>
	<remap from="out" to="/bta_tof_driver_1/tof_camera/decompressedDepth"/>
  </node>
</launch>
